spec/intl.html

```diff
@@ -40,15 +41,92 @@ <h1><a href="https://tc39.es/ecma402/#sec-case-sensitivity-and-case-mapping">Cas
+        These merges can be problematic because geographically and politically distinct locations, especially across country boundaries, are more likely to introduce divergent time zone rules in a future version of the IANA Time Zone Database.
+        Therefore, it is recommended that all identifiers listed in the IANA Time Zone Database's <code>zone.tab</code> file (which lists at least one Link or Zone name for each <a href="https://www.iso.org/glossary-for-iso-3166.html">ISO 3166-1 Alpha-2</a> country code in the IANA Time Zone Database) should be primary identifiers.
+        One way to achieve this result is to build the IANA Time Zone Database with the <code>PACKRATDATA=backzone PACKRATLIST=zone.tab</code> build options.
       </p>
-    </ins>
+    </emu-note>
+
+      <emu-clause id="sup-availablenamedtimezoneidentifiers" oldids="sec-availabletimezones" type="implementation-defined abstract operation">
+        <h1>AvailableNamedTimeZoneIdentifiers ( ): a List of Time Zone Identifier Records</h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>
+            Its result describes all available named time zone identifiers in this implementation, as well as the primary time zone identifier corresponding to each available named time zone identifier.
+            The List is ordered according to the [[Identifier]] field of each Time Zone Identifier Record.
+          </dd>
+          <dt>redefinition</dt>
+          <dd>true</dd>
+        </dl>
+        <p>This definition supersedes the definition provided in <emu-xref href="#sec-availablenamedtimezoneidentifiers"></emu-xref>.</p>
+        <emu-alg>
+          1. Let _identifiers_ be a List containing the String value of each Zone or Link name in the IANA Time Zone Database.
+          1. Assert: No element of _identifiers_ is an ASCII-case-insensitive match for any other element.
+          1. Assert: Every element of _identifiers_ identifies a Zone or Link name in the IANA Time Zone Database.
+          1. Set _identifiers_ to SortStringListByCodeUnit(_identifiers_).
+          1. Let _result_ be a new empty List.
+          1. For each element _identifier_ of _identifiers_, do
+            1. Let _primary_ be _identifier_.
+            1. If _identifier_ is a non-primary time zone identifier and _identifier_ is not *"UTC"*, then
+              1. Set _primary_ to the name of the primary time zone identifier that _identifier_ resolves to, according to the rules for resolving Link names in the IANA Time Zone Database.
+              1. NOTE: An implementation may need to resolve _identifier_ iteratively to obtain the primary time zone identifier.
+            1. If _primary_ is one of *"Etc/UTC"*, *"Etc/GMT"*, or *"GMT"*, set _primary_ to *"UTC"*.
```

Comment:
I think the ECMA-402 override of AvailableNamedTimeZoneIdentifiers can (and should) also be removed by
* moving these three steps into ECMA-262 (with the first one presumably generalized if we intend to allow support for non-IANA zones), and
* updating the above ECMA-402 "Use of the IANA Time Zone Database" section to include normative text constraining implementations' available named time zone identifiers to [full] support of the IANA Time Zone Database.

Reply:
Seems reasonable over the longer term, but not sure we want to bite that off right now, given that it took 2 months to get the 262 text approved. Also it's likely that this AO may change with the canonical-tz proposal, e.g. if it adds Etc/Unknown. Which is another reason that I'd like to wait until the dust settles. Sound OK for now?